paddle/operators/add_op.h

```diff
@@ -29,8 +30,10 @@ class AddKernel : public framework::OpKernel {
 
     output->mutable_data<T>(context.GetPlace());
 
-    output->flat<T>().device(*(context.GetEigenDevice<Place>())) =
-        input0.flat<T>() + input1.flat<T>();
+    framework::EigenVector<T>::Flatten(*output).device(
```

Comment:
Very noising code for implement kernel. Maybe we should think a better way after this PR.

Maybe that way is OK?

```cpp
struct OpKernelUseEigen : framework::OpKernel {
protected:
  template <typename T>
  using Flatten = framework::EigenVector<T>::Flatten;
};

```